Cargo.lock

```diff
@@ -2505,12 +3380,36 @@ dependencies = [
  "rand 0.7.3",
  "scale-info",
  "serde",
- "sp-application-crypto",
- "sp-arithmetic",
- "sp-core",
- "sp-io",
- "sp-std",
- "sp-weights",
+ "sp-application-crypto 6.0.0",
+ "sp-arithmetic 5.0.0",
+ "sp-core 6.0.0",
+ "sp-io 6.0.0",
+ "sp-std 4.0.0",
+ "sp-weights 4.0.0 (git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.32)",
+]
+
+[[package]]
+name = "sp-runtime"
+version = "7.0.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "0edfc5c54c2b31d2f0cf904d472a0bff7125c0c2a2e2330507842e56f9a27444"
```

Comment:
We seem to have duplicate dependencies here. So substrate dependencies come from crates.io and some from github, we should try to get rid of the duplicates.

Reply:
Need to speak with you on this :)

Reply:
What is best approach to go about taking care of this in a methodical way?

Reply:
You can first check:

```bash

 cargo tree -i -p sp-io                                                                                                                ok | 16:59:35
error: There are multiple `sp-io` packages in your project, and the specification `sp-io` is ambiguous.
Please re-run this command with `-p <spec>` where `<spec>` is one of the following:
  sp-io@6.0.0
  sp-io@7.0.0
  
  cargo tree -i -p sp-io@7.0.0                                                                                                     101 err | 16:59:44
sp-io v7.0.0
├── sp-application-crypto v7.0.0
│   └── sp-runtime v7.0.0
│       └── subxt v0.25.0
│           [dev-dependencies]
│           └── pallet-xcm-transactor v0.1.0 (/home/clang/code/integritee-pallets/xcm-transactor)
└── sp-runtime v7.0.0 (*)
```

Now we know that the dependency does indeed come from the subxt dependency (sp-io was a random choice not, but I have gotten used to it because it is the crate that introduces the duplicate rust lang items if you have it twice.)

So much for the identification. Resolving this can either be done by looking for a subxt version, which also fetches the substrate dependencies from github and from the `polakdot-v0.9.32` branch to be precise, which is quite the nuisance actually, or we patch the substrate versions, but this is also quite the nuisance (you can only patch it if the dependences have the same version).

I am not sure what the best approach is currently. We might want to just ignore it, as it is only a dev-dependency which will (hopefully) not pollute dependants.

Reply:
yea I saw on subxt repo that there is a polkadot-v0.9.x.x branch but only for one version of polkadot lol. I guess we just ignore since it is a dev dependency but probably should pay close attention to this in the future